subsys/shell/Kconfig

```diff
@@ -17,6 +17,14 @@ module-str = Shell
 source "subsys/logging/Kconfig.template.log_config"
 source "subsys/shell/Kconfig.backends"
 
+config SHELL_LIGHTWEIGHT
```

Comment:
would SHELL_MINIMAL be a better name?

Reply:
why MINIMAL over LIGHTWEIGHT?

Reply:
Well it seems to be the pattern in Zephyr, e.g. MINIMAL_LIBC

git grep MINIMAL

Reply:
+1 for MINIMAL

e.g. LOG_MODE_MINIMAL

Reply:
Why MINIMAL vs. LIGHTWEIGHT?

Reply:
I see both uses in the Zephyr code, but "minimal" seems to be more widely adopted:  e.g. CONFIG_MINIMAL_LIBC, CONFIG_LOG_MINIMAL, LOG_MODE_MINIMAL.  I vote for SHELL_MINIMAL

Reply:
Done